### PR TITLE
A4A: Sites Dashboard - Preview Pane - Hosting Feature design

### PR DESCRIPTION
**Pull request by gogdzl**
Related to https://github.com/Automattic/automattic-for-agencies-dev/issues/263

## Proposed Changes

* Update the design of the Hosting feature in the preview pane. The design was updated to match the provided Figma file:  1UpoqS1KkCtLXVX68TbMSM-fi
* Change the php_version_num field to a string, since it contains semantic versioning information
* Add new fields to be displayed in the Hosting feature: `wordpress_version` and `hosting_provider_guess`

**Note**: This PR does not update the 'Status' information. Using the `public` field proved to be unreliable. This information will have to be updated in a subsequent PR. Context: https://github.com/Automattic/automattic-for-agencies-dev/issues/264 

## Testing Instructions

* Run `yarn start-a8c-for-agencies`
* Go to `http://agencies.localhost:3000/sites`
* Click over a site to open the preview pane
* The "Hosting" tab should show the design matching the Figma file
* The information should be correct, with the exception of `Status`
* Check if the logo is shown for WordPress.com and Pressable sites

image

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/sections/sites/features/hosting/style.scss [hunk 1]
```diff
@@ -1,3 +1,64 @@
-.site-preview-pane__hosting-content {
-	margin: 32px 0;
+.hosting__card {
```
Comment (cleacos) [inline client/a8c-for-agencies/sections/sites/features/hosting/style.scss]: Could we wrap these classes into `.hosting__wrapper`? That way, we are more specific and avoid potential class name collisions.

File: client/a8c-for-agencies/sections/sites/features/hosting/style.scss [hunk 2]
```diff
@@ -1,3 +1,64 @@
...
+}
+
+.hosting__content-value,
+.hosting__content-label {
+	padding: 12px 24px;
+}
+
+
+.hosting__content-label {
+	color: var(--studio-gray-60, #50575e);
```
Comment (cleacos) [inline client/a8c-for-agencies/sections/sites/features/hosting/style.scss]: Let's use `--color-accent-60`

Reference for the tip: https://github.com/Automattic/wp-calypso/pull/89081/files#r1549034432

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-status-content/hooks/test/use-row-metadata.ts [hunk 3]
```diff
@@ -56,8 +55,16 @@ const FAKE_SITE: Site = {
 		mobile: 10,
 		desktop: 10,
 	},
-	php_version_num: 7.4,
+	php_version_num: '7.4',
```
Comment (cleacos) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-status-content/hooks/test/use-row-metadata.ts]: The `php_version_num` field is a number returned by the endpoint

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-expanded-content/site-php-version.tsx [hunk 4]
```diff
@@ -3,7 +3,7 @@ import { useTranslate } from 'i18n-calypso';
 export default function SitePhpVersion( {
 	phpVersion,
 }: {
-	phpVersion: number | null;
+	phpVersion: string | null;
```
Comment (cleacos) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-expanded-content/site-php-version.tsx]: The `php_version_num` field is a number returned by the endpoint

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/test/test-utils/constants.ts [hunk 5]
```diff
@@ -50,10 +50,17 @@ const site: Site = {
 		mobile: 75,
 		desktop: 85,
 	},
-	php_version_num: 7.4,
-	is_connected: true,
+	php_version_num: '7.4',
```
Comment (cleacos) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/test/test-utils/constants.ts]: The `php_version_num` field is a number returned by the endpoint

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/types.ts [hunk 6]
```diff
@@ -96,7 +96,9 @@ export interface Site {
 	site_stats: SiteStats;
 	onSelect?: () => void;
 	jetpack_boost_scores: BoostData;
-	php_version_num: number;
+	php_version_num: string;
```
Comment (cleacos) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/types.ts]: The `php_version_num` field is a number returned by the endpoint

File: client/a8c-for-agencies/data/sites.ts [hunk 7]
```diff
@@ -26,7 +26,9 @@ export const mockedSites: Site[] = [
 		latest_scan_threats_found: [],
 		latest_scan_has_threats_found: false,
 		jetpack_boost_scores: { desktop: 0, mobile: 0, overall: 0 },
-		php_version_num: 8.1,
+		php_version_num: '8.1',
```
Comment (cleacos) [inline client/a8c-for-agencies/data/sites.ts]: The php_version_num field is a number returned by the endpoint

File: client/a8c-for-agencies/data/sites.ts [hunk 8]
```diff
@@ -78,7 +80,9 @@ export const mockedSites: Site[] = [
...
```
Comment (cleacos) [inline client/a8c-for-agencies/data/sites.ts]: The php_version_num field is a number returned by the endpoint

File: client/a8c-for-agencies/data/sites.ts [hunk 9]
```diff
@@ -138,7 +142,9 @@ export const mockedSites: Site[] = [
 			mobile: 68,
 			overall: 82,
 		},
-		php_version_num: 7.4,
+		php_version_num: '7.4',
```
Comment (cleacos) [inline client/a8c-for-agencies/data/sites.ts]: The php_version_num field is a number returned by the endpoint

File: client/a8c-for-agencies/data/sites.ts [hunk 10]
```diff
@@ -204,7 +210,9 @@ export const mockedSites: Site[] = [
 			mobile: 46,
 			overall: 57,
 		},
-		php_version_num: 7.4,
+		php_version_num: '7.4',
```
Comment (cleacos) [inline client/a8c-for-agencies/data/sites.ts]: The php_version_num field is a number returned by the endpoint

File: client/a8c-for-agencies/sections/sites/features/hosting/overview.tsx [hunk 11]
```diff
@@ -10,19 +12,76 @@ type Props = {
...
+
+		// Currently we are only converting the name from 'snake_case' and 'kebab-case' to 'Title Case'
+		const formattedName = hostProviderGuess
+			.replace( /[-_]/g, ' ' )
+			.split( ' ' )
+			.map( ( word ) => word.charAt( 0 ).toUpperCase() + word.slice( 1 ) )
+			.join( ' ' );
+
+		return formattedName;
+	};
```
Comment (cleacos) [inline client/a8c-for-agencies/sections/sites/features/hosting/overview.tsx:47]: ❤️

**Review by cleacos — COMMENTED** [review]
Great! It looks great

image

**I left some comments with minor issues, if we can fix them. Other than that, I'll be good to merge.**

File: client/a8c-for-agencies/sections/sites/features/hosting/style.scss [hunk 12]
```diff
@@ -1,3 +1,64 @@
...
+	font-size: rem(13px);
+	font-weight: 500;
+	line-height: 20px;
+}
+
+.hosting__content-value {
```
Comment (keoshi) [inline client/a8c-for-agencies/sections/sites/features/hosting/style.scss]: When it's Pressable hosting the icon + label are slightly misaligned.

Can we add `display: flex; align-items: center;` to mitigate this? Thank you!

**Review by keoshi — APPROVED** [review]
Left a tiny comment, but other than that this looks great! 🙌

File: client/a8c-for-agencies/sections/sites/features/hosting/style.scss (hunk 12 above)
Comment (gogdzl) [reply client/a8c-for-agencies/sections/sites/features/hosting/style.scss]: Yes, we can add it: fafa086

File: client/a8c-for-agencies/sections/sites/features/hosting/overview.tsx (hunk 11 above)
Comment (gogdzl) [reply client/a8c-for-agencies/sections/sites/features/hosting/overview.tsx:47]: :)

File: client/a8c-for-agencies/data/sites.ts (hunk 10 above)
Comment (gogdzl) [reply client/a8c-for-agencies/data/sites.ts]: Reverted back to number in fb88090.

File: client/a8c-for-agencies/data/sites.ts (hunk 9 above)
Comment (gogdzl) [reply client/a8c-for-agencies/data/sites.ts]: Reverted back to number in fb88090.

File: client/a8c-for-agencies/data/sites.ts (hunk 8 above)
Comment (gogdzl) [reply client/a8c-for-agencies/data/sites.ts]: Reverted back to number in fb88090.

File: client/a8c-for-agencies/data/sites.ts (hunk 7 above)
Comment (gogdzl) [reply client/a8c-for-agencies/data/sites.ts]: Reverted back to number in fb88090.

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/types.ts (hunk 6 above)
Comment (gogdzl) [reply client/jetpack-cloud/sections/agency-dashboard/sites-overview/types.ts]: Reverted back to number in fb88090.

Comment (gogdzl) [reply client/jetpack-cloud/sections/agency-dashboard/sites-overview/types.ts]: Reverted back to number in fb88090.

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/test/test-utils/constants.ts (hunk 5 above)
Comment (gogdzl) [reply client/jetpack-cloud/sections/agency-dashboard/sites-overview/test/test-utils/constants.ts]: Reverted back to number in fb88090.

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-expanded-content/site-php-version.tsx (hunk 4 above)
Comment (gogdzl) [reply client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-expanded-content/site-php-version.tsx]: Reverted back to number in fb88090.

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-status-content/hooks/test/use-row-metadata.ts (hunk 3 above)
Comment (gogdzl) [reply client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-status-content/hooks/test/use-row-metadata.ts]: Reverted back to number in fb88090.

File: client/a8c-for-agencies/sections/sites/features/hosting/style.scss [hunk 13]
```diff
@@ -1,3 +1,68 @@
...
+	}
+
+	.hosting__content-value,
+	.hosting__content-label {
+		padding: 12px 24px;
+	}
+
+
+	.hosting__content-label {
+		color: var(--studio-gray-60, #50575e);
```
Comment (cleacos) [inline client/a8c-for-agencies/sections/sites/features/hosting/style.scss]:
```suggestion
		color: var(--color-accent-60, #50575e);
```

File: client/a8c-for-agencies/sections/sites/features/hosting/style.scss (hunk 2 above)
Comment (gogdzl) [reply client/a8c-for-agencies/sections/sites/features/hosting/style.scss]: Changed in 2a3941e. Thanks for the reference!

File: client/a8c-for-agencies/sections/sites/features/hosting/style.scss (hunk 1 above)
Comment (gogdzl) [reply client/a8c-for-agencies/sections/sites/features/hosting/style.scss]: Wrapped everything up here: d0bd5a4

**Review by cleacos — APPROVED** [review]
Nice! 

LGTM!

image

image

image

**Pull request merged**